internal/api/http/middleware/access_interceptor.go

```diff
@@ -53,18 +57,29 @@ func (a *AccessInterceptor) Handle(next http.Handler) http.Handler {
 		wrappedWriter := &statusRecorder{ResponseWriter: writer, status: 0}
 
 		instance := authz.GetInstance(ctx)
-		remaining := a.svc.Limit(tracingCtx, instance.InstanceID())
-		limit := remaining != nil && *remaining == 0
-
-		a.cookieHandler.SetCookie(wrappedWriter, a.limitConfig.ExhaustedCookieKey, request.Host, strconv.FormatBool(limit))
-
+		limit := false
+		if !a.storeOnly {
+			remaining := a.svc.Limit(tracingCtx, instance.InstanceID())
+			limit = remaining != nil && *remaining == 0
+		}
 		if limit {
-			wrappedWriter.WriteHeader(http.StatusTooManyRequests)
-			wrappedWriter.ignoreWrites = true
+			// Limit can only be true when storeOnly is false, so set the cookie and the response code
+			cookieValue, err := templateCookieValue(a.limitConfig.ExhaustedCookieValue, instance)
+			if err != nil {
+				// If templating didn't succeed, emit a warning log and just use the plain config
+				logging.WithError(err).WithField("value", a.limitConfig.ExhaustedCookieValue).Warning("failed to go template cookie value config")
+				err = nil
+			}
+			a.cookieHandler.SetCookie(wrappedWriter, a.limitConfig.ExhaustedCookieKey, request.Host, cookieValue)
+			http.Error(wrappedWriter, "quota for authenticated requests is exhausted", http.StatusTooManyRequests)
+		} else {
+			if !a.storeOnly {
+				// If not limited and not storeOnly, ensure the cookie is deleted
+				DeleteExhaustedCookie(a.cookieHandler, wrappedWriter, request, a.limitConfig)
+			}
+			// Always serve if not limited
+			next.ServeHTTP(wrappedWriter, request)
 		}
-
-		next.ServeHTTP(wrappedWriter, request)
-
```

Comment:
This is actually an important fix, because requests should not be served when they are limited